base/expr.jl

```diff
@@ -230,12 +230,22 @@ end
 `@pure` gives the compiler a hint for the definition of a pure function,
 helping for type inference.
 
-A pure function can only depend on immutable information.
-This also means a `@pure` function cannot use any global mutable state, including
-generic functions. Calls to generic functions depend on method tables which are
+The criteria used by Julia to deem a function pure is stricter than the one
+used by most other languages, and incorrect `@pure` annotation may introduce
```

Comment:
"other languages" is a weasel word that means nothing. Our definition of pure is (currently) weaker than LLVM's, for example.

Reply:
Removed the unnecessary comment. Now it is just "very strict".

Reply:
I wonder if instead we should go a slightly different direction here and note that unlike "other languages", we don't require the user to mark a contexprs to tell the Julia compiler something it already knows? So that typically it should not be needed, but that instead using `@pure` is often used as a means of telling the compiler something it knows is false.

This wasn't true a number of years ago, but when we encounter new cases, we tend to try to add them to the compiler instead now.

Reply:
Good thought.  I will make that change.